src/encoded/static/components/matrix.js

```diff
@@ -369,24 +369,58 @@ const convertExperimentToDataTable = (context, getRowCategories, getRowSubCatego
 
 
 /**
- * Render the title panel and list of experiment internal tags.
+ * Display internal tag badges from search results.
+ */
+const MatrixInternalTags = ({ context }) => {
+    // Collect filters that are internal_tags.
+    const internalTags = _.uniq(context.filters.filter(filter => (
+        filter.field === 'internal_tags' && filter.term !== '*'
+    )).map(filter => filter.term));
+    return internalTags.map(tag => (
+        <ImageWithFallback imageUrl={`/static/img/tag-${tag}.png`} imageAlt={`${tag} collection logo`} key={tag} />
+    ));
+};
+
+MatrixInternalTags.propTypes = {
+    /** encode search-results object being displayed */
+    context: PropTypes.object.isRequired,
+};
```

Comment:
One more thought about this: I just remembered I have a ticket to add Internal Tags to search pages, ENCD-4219, so we'll probably actually want this to be called `InternalTags` and move it back to 'objectutils.js'. That can be changed as part of that ticket though. If it were just going to be on the matrix pages this would make the most sense.